src/mesh/serial_mesh.C

```diff
@@ -1271,6 +1271,18 @@ void ReplicatedMesh::stitching_helper (ReplicatedMesh * other_mesh,
               other_mesh->get_boundary_info().boundary_ids(other_elem, s, bc_ids);
               this->get_boundary_info().add_side(this_elem, s, bc_ids);
             }
+
+          // Copy shellface boundary ids
+          // This is only relevant for shell elements
+          if (other_elem->type()==TRISHELL3 || other_elem->type()==QUADSHELL4)
```

Comment:
I would get rid of this "if" statement. Just copy any shellface ids that exist in the original mesh, there's no harm in that, and it's easier to maintain since otherwise if we add new shell types we'd have to remember to update this if-statement.